c7n/resources/guardduty.py

```diff
@@ -0,0 +1,117 @@
+        id = 'InvitationId'
+        name = 'Name'
+        dimension = None
+
+
+@resources.register('guardduty-detectors')
+class GuardDutyDetectors(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'guardduty'
+        enum_spec = ('list_detectors', None, None)
+        id = 'DetectorIds'
+        name = 'DetectorIds'
+        dimension = None
+
+
+@GuardDutyInvitations.filter_registry.register('list-invitations')
+class ListInvitations(ValueFilter):
+    """Filter GuardDuty invitation from trusted Master account id.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: accept-trusted-guardduty-invitation
+                resource: guardduty-invitations
+                filters:
+                  - type: list-invitations
+                    key: AccountId
+                    value: '000000000000'
+    """
+    schema = type_schema('list-invitations', rinherit=ValueFilter.schema)
+    permissions = ('guardduty:ListInvitations',)
+
+    def process(self, resources, event=None):
+        return [r for r in resources if self.match(r)]
+
+
+@GuardDutyDetectors.filter_registry.register('list-detectors')
+class ListDetectors(Filter):
+    schema = type_schema('list-detectors', rinherit=ValueFilter.schema)
+    permissions = ('guardduty:ListDetectors',)
+
+    # Only 1 detector per region is allowed.
+    # So if GuardDuty is enabled then a list containing one element will be returned
+    def process(self, resources, event=None):
+        return resources
+
+
+@GuardDutyInvitations.action_registry.register('accept-invitation')
+class AcceptInvitation(BaseAction):
+    """Filter GuardDuty invitation from trusted Master account id and accept it.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: accept-trusted-guardduty-invitation
+                resource: guardduty-invitations
+                filters:
+                  - type: list-invitations
+                    key: AccountId
+                    value: '000000000000'
+                actions:
+                  - type: accept-invitation
+    """
+    schema = type_schema(
+        'accept-invitation', detectorid={'type': 'string'}, invitationid={'type': 'string'},
+        masterid={'type': 'string'})
+    permissions = ('guardduty:AcceptInvitation',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('guardduty')
+        detectorids = client.list_detectors()['DetectorIds']
+        if len(detectorids) == 0:
+            detectorid = client.create_detector()['DetectorId']
+        else:
+            detectorid = detectorids[0]
+
+        for r in resources:
+            masterid = r['AccountId']
+            invitationid = r['InvitationId']
+            try:
+                client.accept_invitation(DetectorId=detectorid,
+                    InvitationId=invitationid, MasterId=masterid)
+            except Exception as E:
```

Comment:
could you provide more granular error handling here? there are quite a few error cases outlined in the docs: https://docs.aws.amazon.com/guardduty/latest/ug/accept-invitation.html

Reply:
I've added client.exceptions.BadRequestException (specific to GuardDuty client) instead of generic Exception and I'll push new changes for review soon.

Please let me know if there are any examples or conventions to follow here.